lcls-twincat-pmps/PMPS/HelperFunctions/F_SafeBPCompare.TcPOU

```diff
@@ -20,18 +20,26 @@ VAR
     
     idx : UINT;
     xAuxAttOK : BOOL := TRUE;
-    xAuxAprtOK : BOOL := TRUE; 
+    xAuxAprtOK : BOOL := TRUE;
+    
+    Att1 : REFERENCE TO ST_PMPS_Attenuator;
+    Att2 : REFERENCE TO ST_PMPS_Attenuator;
+    
 END_VAR
 ]]></Declaration>
     <Implementation>
       <ST><![CDATA[
-xAttOK := BeamParam1.nTran <= BeamParam2.nTran;
+xAttOK := BeamParam1.nTran <= MIN(PMPS_GVL.TRANS_SCALING_FACTOR, (BeamParam2.nTran *(100+PMPS_PARAM.ATT_MARGIN))/PMPS_GVL.TRANS_SCALING_FACTOR );
 xPPmjOK := BeamParam1.fPP_mJ <= BeamParam2.fPP_mJ;
 xEvOK := (BeamParam1.neVRange AND BeamParam2.neVRange) = BeamParam1.neVRange;
 xRateOK    := BeamParam1.nRate <= BeamParam2.nRate;
 
 FOR idx:=1 to PMPS_GVL.AUX_ATTENUATORS DO
-    xAuxAttOK R= BeamParam1.astAttenuators[idx].nTran > BeamParam2.astAttenuators[idx].nTran;    
+    Att1 REF= BeamParam1.astAttenuators[idx];
+    Att2 REF= BeamParam2.astAttenuators[idx];
+
+    xAuxAttOK R= ( Att1.nTran > MIN( 100, ( (Att2.nTran *(100+PMPS_PARAM.ATT_MARGIN))/100 ) ) )
```

Comment:
some lingering inconsistencies here- where do you pick to use `100` raw vs `PMPS_GVL.TRANS_SCALING_FACTOR`?

Reply:
Yep, you're right this needs to be cleaned up.